feat: Add webhook management functionality to Pipedrive and Zoho CRM API integrations

d-klotz opened the pull request:
This pull request adds and enhances webhook management functionality across the Attio, Pipedrive, and Zoho CRM API client packages. The main focus is to provide methods for creating, listing, updating, and deleting webhooks, as well as defining related types. These changes make it easier to manage webhooks programmatically for each integration.

**Webhook Management Enhancements:**

*Attio API (`packages/v1-ready/attio/api.js`):*
- Added a `webhookById` URL helper and a new `updateWebhook` method for updating webhooks by ID. [[1]](diffhunk://#diff-d545ede19d971e67b942e63c6c0ec89d907e23e833966dfe7db36119b578866bR26) [[2]](diffhunk://#diff-d545ede19d971e67b942e63c6c0ec89d907e23e833966dfe7db36119b578866bR219-R229)

*Pipedrive API (`packages/v1-ready/pipedrive/src/api.ts`):*
- Introduced URL helpers for `webhooks` and `webhookById`. [[1]](diffhunk://#diff-ea29ec09744691639b5cbfc98b585a3e181f9b22ae1e1118d046bff8b0f88fb8R30-R31) [[2]](diffhunk://#diff-ea29ec09744691639b5cbfc98b585a3e181f9b22ae1e1118d046bff8b0f88fb8R53-R54)
- Added methods to list all webhooks, create a new webhook, and delete a webhook by ID, with appropriate request/response typing. [[1]](diffhunk://#diff-ea29ec09744691639b5cbfc98b585a3e181f9b22ae1e1118d046bff8b0f88fb8R14) [[2]](diffhunk://#diff-ea29ec09744691639b5cbfc98b585a3e181f9b22ae1e1118d046bff8b0f88fb8R259-R308)

*Zoho CRM API (`packages/v1-ready/zoho-crm/src/api.ts`, `types.ts`):*
- Added URL helpers for webhooks and methods to list, get, create, update, and delete webhooks, including input validation and error handling. [[1]](diffhunk://#diff-de24aed5a7de6e7c70ef3668d0ca2738ebd559f2c6b227f88bad2f351ff3b81aR53-R54) [[2]](diffhunk://#diff-de24aed5a7de6e7c70ef3668d0ca2738ebd559f2c6b227f88bad2f351ff3b81aR333-R434)
- Defined `WebhookConfig` and `WebhookResponse` interfaces for type safety and clarity in webhook operations. [[1]](diffhunk://#diff-de24aed5a7de6e7c70ef3668d0ca2738ebd559f2c6b227f88bad2f351ff3b81aR17-R18) [[2]](diffhunk://#diff-5bd608adbfb626b09a56372297faed0bca6ff70d071db8caeb124fd5104eb95cR245-R276)
<!-- GITHUB_RELEASE PR BODY: canary-version -->
<details>
  <summary>📦 Published PR as canary version: <code>Canary Versions</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @friggframework/api-module-attio@1.1.0-canary.58.42ea316.0
  npm install @friggframework/api-module-pipedrive@2.1.0-canary.58.42ea316.0
  npm install @friggframework/api-module-zoho-crm@1.1.0-canary.58.42ea316.0
  # or 
  yarn add @friggframework/api-module-attio@1.1.0-canary.58.42ea316.0
  yarn add @friggframework/api-module-pipedrive@2.1.0-canary.58.42ea316.0
  yarn add @friggframework/api-module-zoho-crm@1.1.0-canary.58.42ea316.0
  ```
</details>
<!-- GITHUB_RELEASE PR BODY: canary-version -->

<!-- GITHUB_RELEASE PR BODY: prerelease-version -->
# Version

Published prerelease version: `@friggframework/api-module-attio@1.1.0-next.2`
`@friggframework/api-module-pipedrive@2.1.0-next.2`
`@friggframework/api-module-zoho-crm@2.0.0-next.2`

<details>
  <summary>Changelog</summary>

  #### 🚀 Enhancement
  
  - `@friggframework/api-module-attio`, `@friggframework/api-module-pipedrive`, `@friggframework/api-module-zoho-crm`
    - feat: Add webhook management functionality to Pipedrive and Zoho CRM API integrations [#58](https://github.com/friggframework/api-module-library/pull/58) ([@d-klotz](https://github.com/d-klotz))
  - `@friggframework/api-module-microsoft-teams`, `@friggframework/api-module-slack`, `@friggframework/api-module-42matters`, `@friggframework/api-module-asana`, `@friggframework/api-module-attio`, `@friggframework/api-module-connectwise`, `@friggframework/api-module-contentful`, `@friggframework/api-module-contentstack`, `@friggframework/api-module-crossbeam`, `@friggframework/api-module-deel`, `@friggframework/api-module-frigg-scale-test`, `@friggframework/api-module-frontify`, `@friggframework/api-module-google-calendar`, `@friggframework/api-module-google-drive`, `@friggframework/api-module-helpscout`, `@friggframework/api-module-hubspot`, `@friggframework/api-module-ironclad`, `@friggframework/api-module-linear`, `@friggframework/api-module-pipedrive`, `@friggframework/api-module-salesforce`, `@friggframework/api-module-stripe`, `@friggframework/api-module-unbabel-projects`, `@friggframework/api-module-unbabel`, `@friggframework/api-module-zoho-crm`, `@friggframework/api-module-zoom`
    - refactor: zoho crm api module [#57](https://github.com/friggframework/api-module-library/pull/57) ([@d-klotz](https://github.com/d-klotz))
  
  #### Authors: 1
  
  - Daniel Klotz ([@d-klotz](https://github.com/d-klotz))
</details>
<!-- GITHUB_RELEASE PR BODY: prerelease-version -->

d-klotz committed:
feat: Add webhook management functionality to Pipedrive and Zoho CRM API integrations

seanspeaks commented on packages/v1-ready/zoho-crm/src/api.ts:
```diff
@@ -48,6 +50,8 @@ export class Api extends OAuth2Requester {
             accounts: '/Accounts',
             account: (accountId: string) => `/Accounts/${accountId}`,
             accountSearch: '/Accounts/search',
+            webhooks: '/settings/automation/webhooks',
```
See slack thread- wrong "webhooks". 

This is for Zoho Flow and I think their "automations" tooling inside CRM, i.e. as an end user I can configure a webhook to send data out to (Think: Webhooks by Zapier)

We're looking for "notifications" for the REST API in RM https://www.zoho.com/crm/developer/docs/api/v8/notifications/enable.html

d-klotz committed:
feat: Enhance Pipedrive and Zoho CRM API integrations with new note and search functionalities

- Added methods for creating, updating, retrieving, and deleting notes in the Zoho CRM API.
- Introduced search functionality for persons and general items in the Pipedrive API.
- Updated TypeScript definitions to include new parameters and response types for notes and search operations.

seanspeaks commented:
<!-- GITHUB_RELEASE COMMENT: released -->
:rocket: PR was released in [`@friggframework/api-module-frontify@2.0.0-next.14`](https://github.com/friggframework/api-module-library/releases/tag/%40friggframework/api-module-frontify%402.0.0-next.14) :rocket:
